packages/yew-router/src/utils.rs

```diff
@@ -30,24 +11,39 @@ pub fn fetch_base_url() -> Option<String> {
             let base = url.pathname();
 
             let base = if base != "/" {
-                strip_slash_suffix(&base)
+                base.strip_suffix("/")
+                    .map(|it| it.to_string())
+                    .unwrap_or(base)
             } else {
                 return None;
             };
 
-            Some(base.to_string())
+            Some(base)
         }
         _ => None,
     }
 }
 
+pub fn base_url() -> Option<String> {
+    thread_local! {
+        static BASE_URL: RefCell<Option<Option<String>>> = RefCell::new(None);
```

Comment:
Why `Option<Option<_>>`?

Reply:
The outer `Option` indicates whether the value has been initialized yet. The inner `Option` corresponds to whether a `base` was actually specified.

If you collapse it to a single option, then you make the common case (no "base") slow because it has to check every time.

Reply:
ie. `Some(None)` indicates we already looked for a base and didn't find one.

Reply:
Can we use `Option<String>` and store no base as `/` or use a wrapper type that stores if we've already checked the base? Double `Option<_>` is confusing from a reader's perspective.

Reply:
I put it behind a type alias, LMK if that's clearer.